python/paddle/fluid/tests/unittests/test_eager_deletion_recurrent_op.py

```diff
@@ -0,0 +1,460 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```

Comment:
As comments above, please add corresponding unittests.
1. Run using ParallelExecutor.
2. There are duplicate recurrent ops in the graph, even it is nested inside another recurrent op.
3. There are recurrent ops with gradient and without gradient.
4. Any corner cases that should be concerned.

Reply:
Done